### DISCUSSION TITLE
Paragraph search

### DISCUSSION THREAD

**Discussion by Mano2731995** [pymupdf/PyMuPDF#2044] [Ideas]
Hai jorj. 
How to find out specific paragraph by using pymupdf. 

Requirement: 

1. Search and extract specific paragraph 
2. Apply strikeout annot
 []()![IMG_20221114_101347.jpg](https://user-images.githubusercontent.com/117537492/201577414-599edc72-c4df-4156-a3b7-aaa004c99e6f.jpg)

**Comment by JorjMcKie** [comment]
For point 1 you are on your own.
Use text properties, position on the page, grouping of lines, etc.
Probably, `page.get_text("dict")` is the best extraction variant for this, because it has all the information like font name, font size, properties like bold, text color. Also lines being in one text block are a good candidate for being in the same "paragraph".

Point 2 is easy then.

**Comment by albydeca** [comment]
See [here](https://medium.com/social-impact-analytics/extract-text-from-unsearchable-pdfs-for-data-analysis-using-python-a6a2ca0866dd) for point 1. Uses `pytesseract`
